src/google/protobuf/stubs/platform_macros.h

```diff
@@ -115,6 +115,8 @@ GOOGLE_PROTOBUF_PLATFORM_ERROR
 // TLS support on android.
 // iOS also does not support the __thread keyword.
 #define GOOGLE_PROTOBUF_NO_THREADLOCAL
+
+#include <pthread.h>
```

Comment:
This include is only added if `GOOGLE_PROTOBUF_NO_THREADLOCAL` is defined (i.e., the case where it's needed). I can move it to the files that use `GOOGLE_PROTOBUF_NO_THREADLOCAL`, but there are multiple ones and if a new file used `GOOGLE_PROTOBUF_NO_THREADLOCAL` it would need to add the include.